ui/src/views/Tasks/TaskButtons/index.jsx

```diff
@@ -0,0 +1,751 @@
+import { hot } from 'react-hot-loader';
+import React, { Component, Fragment } from 'react';
+import { withApollo } from 'react-apollo';
+import { withRouter } from 'react-router-dom';
+import { omit } from 'ramda';
+import cloneDeep from 'lodash.clonedeep';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import Typography from '@material-ui/core/Typography';
+import Checkbox from '@material-ui/core/Checkbox';
+import HammerIcon from 'mdi-react/HammerIcon';
+import CreationIcon from 'mdi-react/CreationIcon';
+import PencilIcon from 'mdi-react/PencilIcon';
+import ClockOutlineIcon from 'mdi-react/ClockOutlineIcon';
+import ShovelIcon from 'mdi-react/ShovelIcon';
+import CloseIcon from 'mdi-react/CloseIcon';
+import FlashIcon from 'mdi-react/FlashIcon';
+import OpenInNewIcon from 'mdi-react/OpenInNewIcon';
+import ConsoleLineIcon from 'mdi-react/ConsoleLineIcon';
+import RestartIcon from 'mdi-react/RestartIcon';
+import SpeedDial from '../../../components/SpeedDial';
+import SpeedDialAction from '../../../components/SpeedDialAction';
+import DialogAction from '../../../components/DialogAction';
+import TaskActionForm from '../../../components/TaskActionForm';
+import formatError from '../../../utils/formatError';
+import removeKeys from '../../../utils/removeKeys';
+import parameterizeTask from '../../../utils/parameterizeTask';
+import { nice } from '../../../utils/slugid';
+import { TASK_ADDED_FIELDS } from '../../../utils/constants';
+import formatTaskMutation from '../../../utils/formatTaskMutation';
+import scheduleTaskQuery from './scheduleTask.graphql';
+import rerunTaskQuery from './rerunTask.graphql';
+import cancelTaskQuery from './cancelTask.graphql';
+import purgeWorkerCacheQuery from './purgeWorkerCache.graphql';
+import createTaskQuery from '../createTask.graphql';
+import submitTaskAction from '../submitTaskAction';
+
+@withRouter
+@hot(module)
+@withApollo
+export default class TaskActionButtons extends Component {
```

Comment:
Maybe we can rename this component to `TaskActionsSpeedDial` and move it to be in `src/components`.